test/services/storage/storage.spec.ts

```diff
@@ -0,0 +1,202 @@
+import chai from 'chai'
+import chaiAsPromised from 'chai-as-promised'
+import sinonChai from 'sinon-chai'
+import dirtyChai from 'dirty-chai'
+import sinon from 'sinon'
+import { EventData } from 'web3-eth-contract'
+import { asciiToHex } from 'web3-utils'
+import Eth from 'web3-eth'
+import { SubstituteOf } from '@fluffy-spoon/substitute'
+import { Sequelize } from 'sequelize-typescript'
+
+import eventProcessor from '../../../src/services/storage/storage.processor'
+import { OfferService, StorageServices } from '../../../src/services/storage'
+import { sequelizeFactory } from '../../../src/sequelize'
+import Offer from '../../../src/services/storage/models/offer.model'
+import { eventMock } from '../../utils'
+import { EventError } from '../../../src/errors'
+import BillingPlan from '../../../src/services/storage/models/price.model'
+
+chai.use(sinonChai)
+chai.use(chaiAsPromised)
+chai.use(dirtyChai)
+const expect = chai.expect
+
+describe('Storage services', function () {
+  const provider = 'TestAddress'
+  let sequelize: Sequelize
+  let eth: SubstituteOf<Eth>
+
+  before(() => {
+    sequelize = sequelizeFactory()
+  })
+
+  describe('Events Processor', () => {
+    describe('Offer events', () => {
+      let processor: (event: EventData) => Promise<void>
+      let offerService: OfferService
+      let offerServiceEmitSpy: sinon.SinonSpy
+
+      before(async () => {
+        offerService = new OfferService({ Model: Offer })
+        processor = eventProcessor({ offerService } as StorageServices, eth)
+        await sequelize.sync({ force: true })
+      })
+      beforeEach(() => {
+        offerServiceEmitSpy = sinon.spy()
```

Comment:
I know I have this pattern as well in my tests, so just FYI if you want to be better then me 😅   then there is the [`spy.resetHistory()`](https://sinonjs.org/releases/v9.0.2/spies/#spyresethistory) call so you don't have to recreate and reassign the spy every test :-)